html/user-activation/activation-same-and-cross-origin.sub.html

```diff
@@ -0,0 +1,148 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="/resources/testdriver.js"></script>
+    <script src="/resources/testdriver-vendor.js"></script>
+    <script src="resources/utils.js"></script>
+  </head>
+  <body>
+    <h1>Check that cross origin iframes don't get activated</h1>
+    <p>
+      Tests that activating a same-origin navigable doesn't activate a cross
+      origin navigable.
+    </p>
+    <ol id="instructions">
+      <li>Click inside the yellow area.</li>
+    </ol>
+    <h1>Same origin</h1>
+    <iframe id="so-child" width="200" height="50"></iframe>
+    <h1>Cross origin</h1>
+    <iframe id="xo-child" width="200" height="50"></iframe>
+  </body>
+  <script>
+    const soChild = document.getElementById("so-child");
+    const xoChild = document.getElementById("xo-child");
+
+    function receiveMessage(type) {
+      return new Promise((resolve) => {
+        window.addEventListener("message", function listener(event) {
+          if (typeof event.data !== "string") {
+            return;
+          }
+          const data = JSON.parse(event.data);
+          if (data.type === type) {
+            window.removeEventListener("message", listener);
+            resolve(data);
+          }
+        });
+      });
+    }
+
+    function requestXOReport() {
+      xoChild.contentWindow.postMessage(
+        JSON.stringify({ type: "report" }),
+        "*"
+      );
+      return receiveMessage(`child-two-report`);
+    }
+
+    promise_test(async (t) => {
+      soChild.src = "./resources/child-one.html";
+      xoChild.src =
+        "http://{{hosts[alt][]}}:{{ports[http][0]}}/html/user-activation/resources/child-two.html";
+      const [, unclickedCrossOrigin] = await Promise.all([
+        receiveMessage("child-one-loaded"),
+        receiveMessage("child-two-loaded"),
+      ]);
+      const soActivation = soChild.contentWindow.navigator.userActivation;
+      assert_false(
+        navigator.userActivation.isActive,
+        "navigator.userActivation.isActive must be false"
```

Comment:
Nit: Can we perhaps replace the assert message strings "navigator.userActivation." with "top-frame..." to make them more informative when an assert fails?